packages/core/src/credentials/index.ts

```diff
@@ -0,0 +1,52 @@
+import { JWTPayload, calculateJwkThumbprintUri, exportJWK, SignJWT } from 'jose'
+import ssh2 from 'ssh2'
+import { Buffer } from 'buffer'
+import { KeyObject, createPrivateKey } from 'crypto'
+
+function getAsymmetricKeyAlg(key: KeyObject) {
+  if (!key.asymmetricKeyType) {
+    throw new Error('Only asymmetric keys are supported')
+  }
+  switch (key.asymmetricKeyType) {
+    case 'rsa':
+      return 'RS256'
+    case 'ed25519':
+      return 'EdDSA'
+    default:
+      throw new Error('Only RSA and Ed25519 keys are supported')
+  }
+}
+
+export const jwtGenerator = (privateKey: string | Buffer) => {
+  const sshKey = ssh2.utils.parseKey(privateKey)
+  if (sshKey instanceof Error) {
+    throw new Error(`Could not parse private key: ${sshKey.message}`)
```

Comment:
can throw the returned `Error` (or at least set `cause` to it)

Reply:
done